dynamiqs/integrators/core/save_mixin.py

```diff
@@ -80,3 +89,18 @@ def postprocess_saved(self, saved: Saved, ylast: PyTree) -> Saved:
         # reorder Isave after jax.lax.scan stacking (ntsave, nLm) -> (nLm, ntsave)
         Isave = saved.Isave.swapaxes(-1, -2)
         return eqx.tree_at(lambda x: x.Isave, saved, Isave)
+
+
+class JumpSolveSaveMixin(SolveSaveMixin):
+    """Mixin to assist jump SSE/SME integrators computing time evolution with data
+    saving.
+    """
+
+    def save(self, y: PyTree) -> Saved:
+        # force state normalization before saving
+        return super().save(unit(y))
```

Comment:
Side note @pierreguilmin (unrelated to this PR). If we want to always return normalized states in other solvers, this is quite straightforward with this construction, and we could easily add a tracker for the accumulated norm error too. We might want to generalize this to other solvers.